Source/utils/sdl_wrap.h

```diff
@@ -14,22 +14,55 @@ namespace devilution {
 
 namespace SDLWrap {
 
-inline SDLSurfaceUniquePtr CreateRGBSurface(Uint32 flags, int width, int height, int depth, Uint32 Rmask, Uint32 Gmask, Uint32 Bmask, Uint32 Amask)
+template <typename T>
+T NonNull(T x)
 {
-	SDLSurfaceUniquePtr ret { SDL_CreateRGBSurface(flags, width, height, depth, Rmask, Gmask, Bmask, Amask) };
-	if (ret == nullptr)
+	if (x == nullptr)
 		ErrSdl();
+	return x;
+}
 
-	return ret;
+inline SDLSurfaceUniquePtr CreateRGBSurface(Uint32 flags, int width, int height, int depth, Uint32 Rmask, Uint32 Gmask, Uint32 Bmask, Uint32 Amask)
+{
+	return SDLSurfaceUniquePtr { NonNull(SDL_CreateRGBSurface(flags, width, height, depth, Rmask, Gmask, Bmask, Amask)) };
 }
 
 inline SDLSurfaceUniquePtr CreateRGBSurfaceWithFormat(Uint32 flags, int width, int height, int depth, Uint32 format)
 {
-	SDLSurfaceUniquePtr ret { SDL_CreateRGBSurfaceWithFormat(flags, width, height, depth, format) };
-	if (ret == nullptr)
-		ErrSdl();
+	return SDLSurfaceUniquePtr { NonNull(SDL_CreateRGBSurfaceWithFormat(flags, width, height, depth, format)) };
+}
 
-	return ret;
+inline SDLSurfaceUniquePtr CreateRGBSurfaceWithFormatFrom(void *pixels, int width, int height, int depth, int pitch, Uint32 format)
```

Comment:
Would it be possible to do something similar for the `goto error` in `codec_decode` and `OpenMPQ`?

Reply:
I think your comment was meant for a different PR.

Reply:
No, I'm just asking if you think we could implement a unique pointer to avoid the `goto error` related code in those two function :)

Reply:
I took a look and the short answer is yes:

- For the SHA stuff, the clean solution is to refactor the code and get rid of sgSHA1.
- For the MPQ stuff, I've got a few more commits incoming.

Reply:
Nice, in that case this has my vote for the next thing to tackle in this series :)